src/main/java/seedu/address/model/patient/Record.java

```diff
@@ -0,0 +1,81 @@
+package seedu.address.model.patient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+
+
+/**
+ * Represents a medical record associated with a patient.
+ * Stores information about the patient's visits, initial observations, diagnosis, and treatment plan.
+ */
+public class Record {
+
+    private final Patient patient;
+    private final List<String> departmentsVisited;
```

Comment:
Saw that departments is currently a String type since other members have not implemented the Department class yet, would be helpful if we could create a github issue to integrate it to Department class once its implemented!